Добавлен финальный проект спринта

gorbunov-d opened the pull request.

gorbunov-d committed:
Добавлен финальный проект спринта

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,79 @@
+    collector = BooksCollector()
+    collector.add_new_book('Книга1')
+    collector.add_new_book('Книга2')
+    collector.set_book_genre('Книга1', 'Фантастика')
```
Можно лучше: тут стоит добавлять книги с разными жанрами, чтобы понимать что фильтрация работает корректно

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,79 @@
+    collector.add_new_book('Книга')
+    collector.add_book_in_favorites('Книга')
+    collector.delete_book_from_favorites('Книга')
+    assert 'Книга' not in collector.get_list_of_favorites_books()
```
Нужно исправить: отсутствуют тесты на проверку методов get_book_genre и get_books_genre

gorbunov-d committed:
добавлены:
Улучшенный тест на фильтрацию по жанрам (разные жанры, проверка фильтрации).
Тест на метод get_book_genre (возвращает жанр конкретной книги).
Тест на метод get_books_genre (возвращает словарь всех книг с их жанрами).
